task.go

```diff
@@ -325,7 +325,12 @@ func (t *task) Delete(ctx context.Context, opts ...ProcessDeleteOpts) (*ExitStat
 		return nil, fmt.Errorf("task must be stopped before deletion: %s: %w", status.Status, errdefs.ErrFailedPrecondition)
 	}
 	if t.io != nil {
-		t.io.Close()
+		// io.Wait locks for restored tasks on Windows unless we call
+		// io.Close first (https://github.com/containerd/containerd/issues/5621)
+		// in other cases, preserve the contract and let IO finish before closing
+		if t.client.runtime == fmt.Sprintf("%s.%s", plugin.RuntimePlugin, "windows") {
+			t.io.Close()
+		}
 		t.io.Cancel()
```

Comment:
maybe we should add some comments like "Cancel is used to cancel the goroutine which is still in fifo-opening state (container hasn't been started yet). It's not trying to stop pipe because the pipe should be stopped by shim-side. Otherwise we might loss the data from container"

Reply:
That sounds good! From this and the other PRs, it looks like there's a lot of confusion around what `Cancel` does/should do. I'll add the comment.

Reply:
Thanks @laurazard ! The comment can be handled by the follow-up.

Reply:
Oop, I was a bit too fast, I've added it in this commit. Lmk if you think that looks good @fuweid